File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/Candidate.java
package com.eulersbridge.isegoria.network.api.models;

import com.squareup.moshi.Json;

import org.parceler.Parcel;

@Parcel
public class Candidate {

    @Json(name = "candidateId")
    public long id;
    public long ticketId;
    public long positionId;
    public long userId;

    public String familyName;
    public String givenName;
    public String policyStatement;
    public String information;

    public String getName() {
        return String.format("%s %s", givenName, familyName);
    }

}

File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/Candidate.kt
package com.eulersbridge.isegoria.network.api.models

import android.annotation.SuppressLint
import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@SuppressLint("ParcelCreator")
@Parcelize
data class Candidate (
    @field:Json(name = "candidateId")
    val id: Long = 0,

    val ticketId: Long = 0,
    val positionId: Long = 0,
    val userId: Long = 0,
    val familyName: String?,
    val givenName: String?,
    val policyStatement: String?,
    val information: String?
) : Parcelable {
    val name: String
        get() = "$givenName $familyName"
}